### ISSUE TITLE
Getting blank page with version .8.1

### ISSUE THREAD

**Issue by stupid48** [pantsel/konga#79]
I did a separate install of this version in a completely different root folder from the .6 version.  The server starts up correctly but I get a blank screen when I browse to it.  I get a bunch of javascript errors:

(index):257 Uncaught ReferenceError: angular is not defined
    at (index):257
(anonymous) @ (index):257

I did read another issue where it said to run "bower install" from the app root which I did but it does not seem to resolve the issue.  Any idea?  Thanks, Chris

**Comment by pantsel** [comment]
@stupid48 ,
<pre>Uncaught ReferenceError: angular is not defined</pre>
most probably means that the dependency is not installed.

Check your project's <code>nower_components</code> directory. Is there an <code>angular</code> folder?

Then, in <code>views/layout</code>, where the scripts are included under the SCRIPTS comment. The second one should be 
<pre><script src="bower_components/angular/angular.js?r=0.8.1"></script></pre> and must refer to  a valid path.

If any of the above is not in place, it is clear that the angular dependency is missing. The only thing you can do is a <code>bower --allow-root install</code> in your  project's root folder.

When you do <code>bower --allow-root install</code> manually. Does it pops any errors?

**Comment by stupid48** [comment]
When you say "views/layout", do you mean under the views folder?  I see a file called layout.ejs.  Is that what you mean?

**Comment by pantsel** [comment]
Yes

**Comment by stupid48** [comment]
Whenever I edit that file and add the script src, when I start the server the file gets overwritten and removes the line I just added.

**Comment by pantsel** [comment]
The file's contents are auto-generated when you lift the application. So, can you post the file's contents as they are when you start the app?

**Comment by stupid48** [comment]
Do you mean with the script src entry I added or just as it is by default?

**Comment by pantsel** [comment]
First of all, is there an <code>angular</code> folder in your <code>bower_components</code> with the <code>angular.js</code> file inside?

**Comment by stupid48** [comment]
Yes there is...

**Comment by pantsel** [comment]
OK, and when you run the app, what's the content of the <code>views/layout.ejs</code> file? Can you copy/paste it? The whole file...

**Comment by pantsel** [comment]
Also can you paste the full console output of the blank browser's page?

**Comment by stupid48** [comment]
`<!DOCTYPE html>
<html id="konga">
  <head>
    <title>Konga</title>

    
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

    

    

    
    <link rel="stylesheet" href="styles/importer.css?r=0.8.1">
    
  </head>

  <body data-ng-cloak class="ng-cloak body" ng-controller="MainController">
    <%- body %>
    

    
    
    


    


    <script src="js/dependencies/sails.io.js"></script>

    
    <script src="js/app/apis/00_apis.js?r=0.8.1"></script>
    <script src="js/app/apis/api-model.js?r=0.8.1"></script>
    <script src="js/app/apis/apis-service.js?r=0.8.1"></script>
    <script src="js/app/apis/controllers/add-api-modal-controller.js?r=0.8.1"></script>
    <script src="js/app/apis/controllers/add-api-plugin-modal-controller.js?r=0.8.1"></script>
    <script src="js/app/apis/controllers/api-controller.js?r=0.8.1"></script>
    <script src="js/app/apis/controllers/api-details-controller.js?r=0.8.1"></script>
    <script src="js/app/apis/controllers/api-health-checks-controller.js?r=0.8.1"></script>
    <script src="js/app/apis/controllers/apis-controller.js?r=0.8.1"></script>
    <script src="js/app/apis/views/api-hc-model.js?r=0.8.1"></script>
    <script src="js/app/app.js?r=0.8.1"></script>
    <script src="js/app/certificates/00_certificates.js?r=0.8.1"></script>
    <script src="js/app/certificates/certificate-model.js?r=0.8.1"></script>
    <script src="js/app/certificates/certificates-controller.js?r=0.8.1"></script>
    <script src="js/app/cluster/00_cluster.js?r=0.8.1"></script>
    <script src="js/app/cluster/cluster-model.js?r=0.8.1"></script>
    <script src="js/app/cluster/controllers/cluster-list-controller.js?r=0.8.1"></script>
    <script src="js/app/connections/00_connections.js?r=0.8.1"></script>
    <script src="js/app/connections/connections-controller.js?r=0.8.1"></script>
    <script src="js/app/connections/create-connection-controller.js?r=0.8.1"></script>
    <script src="js/app/connections/node-model.js?r=0.8.1"></script>
    <script src="js/app/connections/update-user-node-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/00_consumers.js?r=0.8.1"></script>
    <script src="js/app/consumers/consumer-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/consumer-model.js?r=0.8.1"></script>
    <script src="js/app/consumers/consumer-service.js?r=0.8.1"></script>
    <script src="js/app/consumers/consumers-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/credentials/consumer-credentials-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/credentials/create-basic-auth-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/credentials/create-hmac-auth-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/credentials/create-jwt-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/credentials/create-key-auth-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/credentials/create-oauth2-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/credentials/mass-assign-credentials-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/details/consumer-details-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/groups/consumer-groups-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/groups/group.js?r=0.8.1"></script>
    <script src="js/app/consumers/groups/manage-groups-modal-controllers.js?r=0.8.1"></script>
    <script src="js/app/consumers/import/import-consumers-connection-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/import/import-consumers-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/import/import-consumers-storage-controller.js?r=0.8.1"></script>
    <script src="js/app/consumers/import/import-consumers-table-controller.js?r=0.8.1"></script>
    <script src="js/app/core/auth/auth.js?r=0.8.1"></script>
    <script src="js/app/core/auth/login/login.js?r=0.8.1"></script>
    <script src="js/app/core/auth/services/00_services.js?r=0.8.1"></script>
    <script src="js/app/core/auth/services/AuthService.js?r=0.8.1"></script>
    <script src="js/app/core/auth/services/UserService.js?r=0.8.1"></script>
    <script src="js/app/core/auth/signup/signup.js?r=0.8.1"></script>
    <script src="js/app/core/components/components.js?r=0.8.1"></script>
    <script src="js/app/core/components/FocusOn.js?r=0.8.1"></script>
    <script src="js/app/core/constants/AccessLevels.js?r=0.8.1"></script>
    <script src="js/app/core/constants/BackendConfig.js?r=0.8.1"></script>
    <script src="js/app/core/core.js?r=0.8.1"></script>
    <script src="js/app/core/dependencies/dependencies.js?r=0.8.1"></script>
    <script src="js/app/core/directives/directives.js?r=0.8.1"></script>
    <script src="js/app/core/directives/konga-loader.js?r=0.8.1"></script>
    <script src="js/app/core/directives/ListSearch.js?r=0.8.1"></script>
    <script src="js/app/core/directives/RawView.js?r=0.8.1"></script>
    <script src="js/app/core/error/error.js?r=0.8.1"></script>
    <script src="js/app/core/filters/filters.js?r=0.8.1"></script>
    <script src="js/app/core/interceptors/00_interceptors.js?r=0.8.1"></script>
    <script src="js/app/core/interceptors/AuthInterceptor.js?r=0.8.1"></script>
    <script src="js/app/core/interceptors/CsrfInterceptor.js?r=0.8.1"></script>
    <script src="js/app/core/interceptors/ErrorInterceptor.js?r=0.8.1"></script>
    <script src="js/app/core/interceptors/HttpTimeoutInterceptor.js?r=0.8.1"></script>
    <script src="js/app/core/interceptors/KongaIntercenptor.js?r=0.8.1"></script>
    <script src="js/app/core/interceptors/TemplateCacheInterceptor.js?r=0.8.1"></script>
    <script src="js/app/core/layout/00_layout.js?r=0.8.1"></script>
    <script src="js/app/core/layout/01_layout-controllers.js?r=0.8.1"></script>
    <script src="js/app/core/layout/02_layout-directives.js?r=0.8.1"></script>
    <script src="js/app/core/layout/03_layout-services.js?r=0.8.1"></script>
    <script src="js/app/core/libraries/libraries.js?r=0.8.1"></script>
    <script src="js/app/core/libraries/LoDash.js?r=0.8.1"></script>
    <script src="js/app/core/models/00_models.js?r=0.8.1"></script>
    <script src="js/app/core/models/DataModel.js?r=0.8.1"></script>
    <script src="js/app/core/plugins/angular-vertilize.js?r=0.8.1"></script>
    <script src="js/app/core/plugins/jsonc.js?r=0.8.1"></script>
    <script src="js/app/core/plugins/material.spinner.js?r=0.8.1"></script>
    <script src="js/app/core/plugins/uiBreadcrumbs.js?r=0.8.1"></script>
    <script src="js/app/core/services/00_services.js?r=0.8.1"></script>
    <script src="js/app/core/services/DataService.js?r=0.8.1"></script>
    <script src="js/app/core/services/DialogService.js?r=0.8.1"></script>
    <script src="js/app/core/services/HttpStatusService.js?r=0.8.1"></script>
    <script src="js/app/core/services/KongPluginsService.js?r=0.8.1"></script>
    <script src="js/app/core/services/ListConfigService.js?r=0.8.1"></script>
    <script src="js/app/core/services/MessageService.js?r=0.8.1"></script>
    <script src="js/app/core/services/NodesService.js?r=0.8.1"></script>
    <script src="js/app/core/services/NotificationsService.js?r=0.8.1"></script>
    <script src="js/app/core/services/RemoteStorageService.js?r=0.8.1"></script>
    <script src="js/app/core/services/Semver.js?r=0.8.1"></script>
    <script src="js/app/core/services/SocketHelperService.js?r=0.8.1"></script>
    <script src="js/app/core/services/SubscriptionsService.js?r=0.8.1"></script>
    <script src="js/app/dashboard/01_dashboard.js?r=0.8.1"></script>
    <script src="js/app/dashboard/02_dashboard-controller.js?r=0.8.1"></script>
    <script src="js/app/info/00_info.js?r=0.8.1"></script>
    <script src="js/app/info/01_info-controllers.js?r=0.8.1"></script>
    <script src="js/app/info/02_info-service.js?r=0.8.1"></script>
    <script src="js/app/plugins/00_plugins.js?r=0.8.1"></script>
    <script src="js/app/plugins/add-plugin-controller.js?r=0.8.1"></script>
    <script src="js/app/plugins/add-plugins-controller.js?r=0.8.1"></script>
    <script src="js/app/plugins/api-plugins-controller.js?r=0.8.1"></script>
    <script src="js/app/plugins/edit-plugin-controller.js?r=0.8.1"></script>
    <script src="js/app/plugins/plugin-helper-service.js?r=0.8.1"></script>
    <script src="js/app/plugins/plugin-model.js?r=0.8.1"></script>
    <script src="js/app/plugins/plugins-controller.js?r=0.8.1"></script>
    <script src="js/app/plugins/plugins-service.js?r=0.8.1"></script>
    <script src="js/app/settings/01_settings.js?r=0.8.1"></script>
    <script src="js/app/settings/email-transport-model.js?r=0.8.1"></script>
    <script src="js/app/settings/settings-controller.js?r=0.8.1"></script>
    <script src="js/app/settings/settings-model.js?r=0.8.1"></script>
    <script src="js/app/settings/settings-service.js?r=0.8.1"></script>
    <script src="js/app/snapshots/00_snapshots.js?r=0.8.1"></script>
    <script src="js/app/snapshots/snapshot-controller.js?r=0.8.1"></script>
    <script src="js/app/snapshots/snapshot-model.js?r=0.8.1"></script>
    <script src="js/app/snapshots/snapshots-controller.js?r=0.8.1"></script>
    <script src="js/app/snapshots/snapshots-service.js?r=0.8.1"></script>
    <script src="js/app/upstreams/01_upstreams.js?r=0.8.1"></script>
    <script src="js/app/upstreams/add-upstream-modal-controller.js?r=0.8.1"></script>
    <script src="js/app/upstreams/edit-upstream-controller.js?r=0.8.1"></script>
    <script src="js/app/upstreams/edit-upstream-details-controller.js?r=0.8.1"></script>
    <script src="js/app/upstreams/targets/edit-upstream-targets-controller.js?r=0.8.1"></script>
    <script src="js/app/upstreams/update-upstream-modal-controller.js?r=0.8.1"></script>
    <script src="js/app/upstreams/upstream.js?r=0.8.1"></script>
    <script src="js/app/upstreams/upstreams-controller.js?r=0.8.1"></script>
    <script src="js/app/upstreams/upstreams-service.js?r=0.8.1"></script>
    <script src="js/app/users/00_users.js?r=0.8.1"></script>
    <script src="js/app/users/user-controller.js?r=0.8.1"></script>
    <script src="js/app/users/user-create-controller.js?r=0.8.1"></script>
    <script src="js/app/users/user-model.js?r=0.8.1"></script>
    <script src="js/app/users/users-controller.js?r=0.8.1"></script>
    

  <script>
    angular.element(document).ready(function() {

      var initInjector = angular.injector(["ng"]);
      var $http = initInjector.get("$http");

      // Load initial configuration to window
      $http.get('api/settings')
              .then(function(response){
                window.KONGA_CONFIG = response.data.data
                window.KONGA_CONFIG_ID = response.data.id
                angular.bootstrap(document, ['frontend']);
              })
              .catch(function(err){
                alert("Error getting initial configuration")
              })


    });
  </script>

  </body>
</html>
`

**Comment by stupid48** [comment]
Sorry, didn't get the whole thing.  Let me try again.

**Comment by stupid48** [comment]
[layout.txt](https://github.com/pantsel/konga/files/1143193/layout.txt)

[console.txt](https://github.com/pantsel/konga/files/1143200/console.txt)

**Comment by pantsel** [comment]
The layout is not generated correctly....
That is strange. I just installed it myself and it works fine.
Can you paste the contents of <code>tasks/pipeline.js</code> as well?

**Comment by pantsel** [comment]
How do you start the app?

**Comment by stupid48** [comment]
I've tried "npm start" and ./start.sh

[pipeline.txt](https://github.com/pantsel/konga/files/1143253/pipeline.txt)

**Comment by pantsel** [comment]
The pipeline file is ok. It should be working. And you said that the app starts with no problems? No wierd warnings or errors when running <code>npm start</code>?

**Comment by stupid48** [comment]
It seems to stat up fine.  No errors or warnings.

**Comment by pantsel** [comment]
I can't think of a reason the <code>views/layout.ejs</code> is not generated properly and I can't reproduce the issue on my end.

Maybe try to clone the project again from scratch?

**Comment by stupid48** [comment]
Sure.  Are the install instructions for version .8 the same as in the documentation .6?  I've been following the .6 doc as I can't find any other place for the current version.

**Comment by stupid48** [comment]
By the way, the tree I'm using is the source code zip file.  Is that correct?

**Comment by pantsel** [comment]
It's better to clone the project with git.

<pre>
$ git clone https://github.com/pantsel/konga.git
$ cd konga
$ npm install
$ npm start
</pre>

**Comment by stupid48** [comment]
OK, I just tried that.  Same issue.  Should I try doing bower --allow-root install?

**Comment by stupid48** [comment]
Actually, that worked after I did bower --allow-root install.  It seems ok now.  I'm not sure what I did wrong the first few times I tried it.  But anyway, it works now.  Thanks very much for your help...

**Comment by pantsel** [comment]
Anytime!
